salt/modules/gpg.py

```diff
@@ -214,6 +215,38 @@ def _search_keys(text, keyserver, user=None):
     return _keys
 
 
+def _gen_key_input(**kwargs):
+    """
+    Generate --gen-key input per gpg doc/DETAILS
+
+    This code is from python-gnupg 0.4.7dev0 it's here because Saltstack
```

Comment:
If this code is from another project it would need to go into salt/ext, with proper licensing and credit attached to it, but I would prefer not to maintain upstream code unless there is very good reason to. If you need to allow a newer version of gnupg that is okay if its not available by default on all operating systems, because they can still install it from pypi if needed. I would prefer not to include this and instead import the module and use the code from the upstream.

Reply:
I picked this snippet from the development unreleased code, the important change is the line '%no-protection' before the '%commit'. It seemed easier to do use the code, but it's possible to use the default gen_input_parameters, check the output for the '%no-protection' when the passphrase is empty and edit the output accordingly.

Reply:
@Ch3LL , I removed the external code and adds the '%no-protection' in the output of `gpg.gen_key_input` call if needed.

The tests are running fine, except for Ubuntu 16.04 which doesn't collect enough entropy to create the keys, but I guess the same can happen with other systems so I'll do an additional commit marking the specific test as slow.

Provided enough entropy the key generation itself is Ok in Ubuntu 16.04:
```
# salt-call grains.get osfinger
local:
    Ubuntu-16.04
# salt-call gpg.create_key use_passphrase=False
local:
    ----------
    fingerprint:
        23037AEB73E710D964E4C113CFA232687258F0FA
    message:
        GPG key pair successfully generated.
    res:
        True
```

Reply:
@Ch3LL I guess the solicited change was addressed, can you check if it's OK now?